pkg/ccl/backupccl/generative_split_and_scatter_processor.go

```diff
@@ -111,7 +114,9 @@ func (n noopSplitAndScatterer) split(_ context.Context, _ keys.SQLCodec, _ roach
 func (n noopSplitAndScatterer) scatter(
 	_ context.Context, _ keys.SQLCodec, _ roachpb.Key,
 ) (roachpb.NodeID, error) {
-	return n.scatterNode, nil
```

Comment:
to be defend against the case where the job coordinator is on an old cluster version and the processor is on a new version, could you return `0, nil` if len(n.sqlInstanceIDs)==0? I don't think this can happen, but better safe than sorry.